File: platform/lang-impl/src/com/intellij/codeInsight/completion/PriorityWeigher.java
package com.intellij.codeInsight.completion;

import org.jetbrains.annotations.NotNull;
import com.intellij.codeInsight.lookup.LookupElement;
import com.intellij.codeInsight.lookup.LookupItem;

/**
 * @author peter
 */
public class PriorityWeigher extends CompletionWeigher {
  @Override
  public Double weigh(@NotNull LookupElement element, CompletionLocation location) {
    final PrioritizedLookupElement prioritized = element.as(PrioritizedLookupElement.class);
    if (prioritized != null) {
      return -prioritized.getPriority();
    }

    final LookupItem item = element.as(LookupItem.class);
    if (item != null) {
      return -item.getPriority();
    }
    return -0.0;
  }
}

File: plugins/kotlin/completion/impl-k2/src/org/jetbrains/kotlin/idea/completion/impl/k2/weighers/PriorityWeigher.kt
// Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.

package org.jetbrains.kotlin.idea.completion.impl.k2.weighers

import com.intellij.codeInsight.lookup.LookupElement
import org.jetbrains.kotlin.idea.completion.ItemPriority
import org.jetbrains.kotlin.idea.completion.priority

internal object PriorityWeigher : KotlinLookupElementWeigher("kotlin.priority") {

    override fun weigh(element: LookupElement): ItemPriority = element.priority ?: ItemPriority.DEFAULT
}